base-compat/src/Data/Tuple/Compat.hs

```diff
@@ -18,3 +21,9 @@ import Data.Tuple
 #if !(MIN_VERSION_base(4,16,0)) && MIN_VERSION_ghc_prim(0,7,0)
 import GHC.Tuple (Solo(..))
 #endif
+
+#if MIN_VERSION_ghc_prim(0,7,0) && !(MIN_VERSION_ghc_prim(0,10,0))
+pattern MkSolo :: a -> Solo a
```

Comment:
luckily patterns don't have identity!

Reply:
Hm, this is a fair point. I think that they _do_ have identity in the sense that the `MkSolo` in `base-compat` would clash with the `MkSolo` in `OneTuple`. In that case, I think the only reasonable thing to do is to not export `MkSolo` on versions where `base` doesn't provide it. That's the only reasonable stance given `base-compat`'s zero-dependencies policy.

Reply:
I've removed this pattern synonym.